articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,212 @@
+ 
+```python
+all_data = get_all_data()
+```
+ 
+Next, we create a checkbox using the *st.checkbox* function that will display the dataset, and also the checkbox should have the values from the *all_data* variable we created earlier.
+ 
+```python
+if st.checkbox('Show Dataset'):
+ st.write(all_data)
+```
+ 
+Now we write another function that will carry out data preprocessing and to achieve that we use the code below.
+ 
+```python
+@st.cache
+def preprocessing_data(data):
+ processing_data = []
+ for single_data in data:
+     if len(single_data.split("\t")) == 2 and single_data.split("\t")[1] != "":
+         processing_data.append(single_data.split("\t"))
+ return processing_data
+ 
+```
+ 
+Now let’s create another checkbox that will display *show_preprocessed_data* and will get the data from the *all_data* function and pass it through the preprocessing data function we wrote above.
+ 
+```python
+if st.checkbox('Show PreProcessed Dataset'):
+ st.write(preprocessing_data(all_data))
+```
+ 
+The next step will be writing a function that will split the data and call it *split_data*
+ 
+```python
+@st.cache
+ 
+def split_data(data):
+ total = len(data)
+ training_ratio = 0.75
+ training_data= []
+ evaluation_data = []
+ for indice in range(0,total):
+     if indice<total*training_ratio:
+         training_data.append(data[indice])
+     else:
+         evaluation_data.append(data[indice])
+ return training_data, evaluation_data
+ 
+```
+ 
+Now we are going to create two functions, one will be *preprocessing_step* that will get the data, preprocess it, and then split it, and the second one will be *training_step* that will receive two parameters and vectorize the training text.
+ 
+```python
+@st.cache
+def preprocessing_step()
+ data = get_all_data()
+ processing_data = preprocessing_data(data)
+ return split_data(processing_data)
+def training_step(data,vectorizer):
+ training_text = [data[0] for data in data]
+ training_result = [data[1] for data in data]
+ training_text = vectorizer.fit_transform(training_text)
+ return BernoulliNB().fit(training_text,training_result)
+```
+ 
+The next step will be passing the training and evaluation data into the *preprocessing_step* function, choosing a vectorizer( In this case we will be using CountVecotizer), and finally, a classifier variable that will call the *training_step* function and pass in the training_data and vectorizer. And the last line uses BernoulliNB which is a Naives Bayes model it predicts the probability of the input being classified for all the classes and is used for text classification with the 'Bag of words' model, we then fit our *training_text* and *training_result* into it.
```

Comment:
Break it down. Also... while mentioning variables using \`\` symbol. Don't use * * to highlight variables.